lms/djangoapps/badges/api/views.py

```diff
@@ -0,0 +1,135 @@
+"""
+API views for badges
+"""
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import generics
+from rest_framework.exceptions import APIException
+
+from openedx.core.djangoapps.user_api.permissions import is_field_shared_factory
+from openedx.core.lib.api.authentication import (
+    OAuth2AuthenticationAllowInactiveUser,
+    SessionAuthenticationAllowInactiveUser
+)
+from badges.models import BadgeAssertion
+from .serializers import BadgeAssertionSerializer
+from xmodule_django.models import CourseKeyField
+
+
+class CourseKeyError(APIException):
+    """
+    Raised the course key given isn't valid.
+    """
+    status_code = 400
+    default_detail = "The course key provided could not be parsed."
```

Comment:
Is a parsing issue the only reason we'd be returning this 400 response?  It might be better to simply say that the course key is invalid.